sweepbatcher/presigned.go

```diff
@@ -593,23 +596,30 @@ func CheckSignedTx(unsignedTx, signedTx *wire.MsgTx, inputAmt btcutil.Amount,
 	}
 
 	// Compare outputs.
-	if len(unsignedTx.TxOut) != 1 {
-		return fmt.Errorf("unsigned tx has %d outputs, want 1",
-			len(unsignedTx.TxOut))
-	}
-	if len(signedTx.TxOut) != 1 {
-		return fmt.Errorf("the signed tx has %d outputs, want 1",
+	if len(unsignedTx.TxOut) != len(signedTx.TxOut) {
+		return fmt.Errorf("unsigned tx has %d outputs, signed tx has "+
+			"%d outputs, should be equal", len(unsignedTx.TxOut),
 			len(signedTx.TxOut))
 	}
-	unsignedOut := unsignedTx.TxOut[0]
-	signedOut := signedTx.TxOut[0]
-	if !bytes.Equal(unsignedOut.PkScript, signedOut.PkScript) {
-		return fmt.Errorf("mismatch of output pkScript: %x, %x",
-			unsignedOut.PkScript, signedOut.PkScript)
+	for i, o := range unsignedTx.TxOut {
+		if !bytes.Equal(o.PkScript, signedTx.TxOut[i].PkScript) {
+			return fmt.Errorf("mismatch of output pkScript: %x, %x",
+				o.PkScript, signedTx.TxOut[i].PkScript)
+		}
+	}
+
+	// The first output is always the batch output.
+	batchOutput := signedTx.TxOut[0]
+
+	// Calculate the total value of change outputs to help determine the
+	// transaction fee.
+	totalChangeValue := btcutil.Amount(0)
+	for i := 1; i < len(signedTx.TxOut); i++ {
+		totalChangeValue += btcutil.Amount(signedTx.TxOut[i].Value)
```

Comment:
We are not interested in `batchOutput` itself, it is needed only to calculate the fee. This code can be simplified: just decrease inputAmt by the sum of all outputs and it will be the fee.